### PR TITLE
msglist: Don't assume people with the same name are the same.

### PR DESCRIPTION
**Pull request by gnprice**
Fixes: #4303

Sadly we can't use the sender's actual, permanent, numeric user ID here, because outbox messages (values of type Outbox) don't have a sender_id. For now, using the email suffices to fix this bug, and leaves this code no worse than a lot of the rest of our code.

We also add a sender_id property to Outbox... but don't yet switch to using it, because existing outbox messages, if there were any just before the user upgrades to a release with this change, won't yet have it. Once such a release has been out for a few weeks, we can make the property required and have a migration just drop any values lacking it. There's a TODO comment marking this as one of the many sites of #3674.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
> There's a TODO comment marking this as one of the many sites of #3674.

Do you mean https://github.com/zulip/zulip-mobile/issues/3764?

File: src/message/renderMessages.js [hunk 1]
```diff
@@ -31,11 +31,11 @@ export default (
         data: [],
       });
     }
+
+    // TODO(#3674): Use sender_id, not sender_email.  Needs adding
```
Comment (chrisbobbe) [inline src/message/renderMessages.js]: I think you mean #3764?

**Comment by chrisbobbe** [comment]
LGTM except for one tiny comment, above—looks like a typo in an issue number.

**Comment by gnprice** [comment]
> Do you mean #3764?

Ha, indeed I do. Thanks for the catch!

I'll edit that and merge.

**Pull request merged**